src/applications/disability-benefits/all-claims/components/ITFBanner.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  itfMessage,
+  itfError,
+  itfSuccess,
+  itfActive,
+} from '../content/itfWrapper';
+
+export default class ITFBanner extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { messageDismissed: false };
+  }
+
+  dismissMessage = () => {
+    this.setState({ messageDismissed: true });
+  };
+
+  render() {
+    if (this.state.messageDismissed) {
+      return <div>{this.props.children}</div>;
+    }
+
+    let message;
+    switch (this.props.status) {
+      case 'error':
+        message = itfMessage(
+          'We’re sorry. Something went wrong on our end.',
+          itfError,
+          'error',
+        );
+        break;
+      case 'itf-found':
+        message = itfMessage(
+          'You already have an Intent to File',
+          itfActive(this.props.currentExpDate),
+          'success',
+        );
+        break;
+      case 'itf-created': {
+        const { previousITF, currentExpDate, previousExpDate } = this.props;
+        message = itfMessage(
+          'Your Intent to File request has been submitted',
+          itfSuccess(previousITF, currentExpDate, previousExpDate),
+          'success',
+        );
+        break;
+      }
+      default:
+        throw new Error(
+          `Unexpected status prop in ITFBanner: ${this.props.status}`,
```

Comment:
Will this get picked up in Sentry automatically?

Reply:
Yeah, I think all uncaught errors will, but the point here was more to catch a misuse of the component.